src/core/aws-net-websocket-registry.adb

```diff
@@ -482,7 +487,9 @@ package body AWS.Net.WebSocket.Registry is
          Socket.On_Close (Message);
          Socket.Shutdown;
 
-         Unchecked_Free (W);
+         if W /= null then
+            DB.Free_Or_Defer (W);
```

Comment:
It would seem more usual for Free_Or_Defer to accept null (and do nothing), like unchecked_deallocation does. But this is very minor